### PR TITLE
APP-2115 - cloud_package_manager improvements

### PR DESCRIPTION
**Pull request by zaporter-work**
Ticket: https://viam.atlassian.net/browse/APP-2115

This PR:
- Expands on package logic to support preserving the permissions of the original tarball (useful for modules).
- Fixes a bug with single-file tarballs in a subdirectory
- Adds the platform field to the agent info

### CODE REVIEW

File: config/reader.go [hunk 1]
```diff
@@ -39,13 +39,15 @@ func getAgentInfo() (*apppb.AgentInfo, error) {
...
 	}
+	platform := fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH)
 
 	return &apppb.AgentInfo{
 		Host:        hostname,
 		Ips:         ips,
 		Os:          runtime.GOOS,
 		Version:     Version,
 		GitRevision: GitRevision,
+		Platform:    &platform,
```
Comment (zaporter-work) [inline config/reader.go:50]: The rdk does not have tests for `AgentInfo`.

Happy to add some if others think it is worthwhile.

**Review by Otterverse — APPROVED** [review]
LGTM

**Review by agreenb — APPROVED** [review]
LGTM

**Review by michaellee1019 — APPROVED** [review]
LGTM!

File: config/reader.go (hunk 1 above)
Comment (zaporter-work) [reply config/reader.go:50]: https://viam.atlassian.net/browse/APP-2276

**Pull request merged**